python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py

```diff
@@ -127,62 +144,69 @@ def execute(self, filename):
+    ``script/ingestRaws.py`` from ``lsst.obs.base`` package.
     """
 
     def __init__(self, config):
-        required = {"root", "instrument"}
-        missing = required - set(config)
+        # Set default values for configuration settings, but override their
+        # them with the ones provided at runtime, if necessary.
+        #
+        # As parallel ingestion is disabled, ``processes`` setting will be
+        # ignored and is included only for sake of completeness.
+        self.config = {
+            "config": None,
+            "config_file": None,
+            "ingest_task": "lsst.obs.base.RawIngestTask",
+            "output_run": None,
+            "processes": 1,
+            "transfer": "symlink",
+        }
+        self.config.update(config)
+
+        required = {"root"}
+        missing = required - set(self.config)
         if missing:
             msg = f"invalid configuration: {', '.join(missing)} not provided"
             logger.error(msg)
             raise ValueError(msg)
-        root = config["root"]
 
-        name = config["instrument"]
-        try:
-            inst = base.utils.getInstrument(name)
-        except (RuntimeError, TypeError) as ex:
-            raise RuntimeError(ex)
-        else:
-            run = inst.makeDefaultRawIngestRunName()
-
-        # Create a Butler.
-        opts = dict(run=run, writeable=True)
-        btl = butler.Butler(root, **opts)
-
-        # Make an attempt to register an instrument with Butler. Quietly
-        # ignore the Butler throwing a fit as most likely it means the
-        # instrument is already registered.
-        #
-        # Note:
-        # We're casting a wide net here as the exception most likely will
-        # depend on the database back-end used by Butler. For example,
-        # for SQLite it look like:
-        #
-        #   sqlite3.IntegrityError: UNIQUE constraint failed: instrument.name
-        try:
-            inst.register(btl.registry)
-        except Exception as ex:
-            logger.debug(f"failed to register the instrument {name}: {ex}")
-            pass
-
-        cfg = base.RawIngestConfig()
-        cfg.transfer = config.get("transfer", "symlink")
-        self.task = base.RawIngestTask(config=cfg, butler=btl)
-
-        pkg = importlib.import_module(ingest.__package__)
+        # Initialize LSST ingest software.
+        butler = Butler(self.config["root"], writeable=True)
+        class_ = doImport(self.config["ingest_task"])
+        config = class_.ConfigClass()
+        config.transfer = self.config["transfer"]
+        config_overrides = ConfigOverrides()
+        if self.config["config_file"] is not None:
+            config_overrides.addFileOverride(self.config["config_file"])
+        if self.config["config"] is not None:
+            for key, val in config.items():
+                config_overrides.addValueOverride(key, val)
+        self.task = class_(config=config, butler=butler)
```

Comment:
Needs "config_overrides.applyTo(config)"

Reply:
Indeed! Thanks for catching it.